### PR TITLE
Add ``--tags`` argument to ``train.py`` and tag run with the sb3 version

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

https://github.com/DLR-RM/rl-baselines3-zoo/pull/335#issuecomment-1366585087
https://github.com/DLR-RM/rl-baselines3-zoo/issues/322#issuecomment-1357812308

```shell
python train.py --track --wandb-entity <ENTITY> --wandb-project-name <PROJECT_NAME>
```

Screenshot 2022-12-28 at 15 05 53

```shell
python train.py --track --wandb-entity <ENTITY> --wandb-project-name <PROJECT_NAME> -tags my-tag
```

Screenshot 2022-12-28 at 15 08 52

```shell
python train.py --track --wandb-entity <ENTITY> --wandb-project-name <PROJECT_NAME> -tags my-tag my-other-tag
```

EDIT: from `--tags 'my-tag, my-other-tag'` to `--tags my-tag my-other-tag`, after this comment https://github.com/DLR-RM/rl-baselines3-zoo/pull/336#discussion_r1058881644

EDIT2: from `--tags` to `-tags` (short for `--wandb-tags`) , after this comment https://github.com/DLR-RM/rl-baselines3-zoo/pull/336#discussion_r1059924739

Screenshot 2022-12-28 at 15 10 03

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
Too add tag to existing runs:

```python
import urllib.request
import wandb

runs = wandb.Api().runs("openrlbenchmark/sb3")

for run in runs:
    url = f"https://api.wandb.ai/files/openrlbenchmark/sb3/{run.id}/requirements.txt"
    try:
        with urllib.request.urlopen(url) as f:
            for line in f:
                line = line.decode("UTF-8")
                if "stable-baselines3==" in line:
                    version = line.strip().split("==")[1]
            if version is None:
                raise Exception("version not found")
            if f"v{version}" not in run._attrs["tags"]:
                run._attrs["tags"].append(f"v{version}")
                run.update()
    except Exception as err:
        print(run.id, err)
```

File: rl_zoo3/train.py [hunk 1]
```diff
@@ -152,6 +153,12 @@ def train():
         default=False,
         help="if toggled, display a progress bar using tqdm and rich",
     )
+    parser.add_argument(
```
Comment (araffin) [inline rl_zoo3/train.py:156]: quick remark (will do full review next week): why not use `nargs="+"`?
so you can do `--tags tag1 tag-2 tag3` and not worry about comma and quotes

Comment (qgallouedec) [reply rl_zoo3/train.py:156]: Indeed, it is better, I did not know this feature.

File: rl_zoo3/train.py [hunk 2]
```diff
@@ -152,6 +153,7 @@ def train():
         default=False,
         help="if toggled, display a progress bar using tqdm and rich",
     )
+    parser.add_argument("--tags", type=str, default="", nargs="+", help="Tags for wandb run, e.g.: --tags optimized pr-123")
```
Comment (araffin) [inline rl_zoo3/train.py:156]:
```suggestion
    parser.add_argument("--tags", type=str, default=[], nargs="+", help="Tags for wandb run, e.g.: --tags optimized pr-123")
```

I would also think renaming it `-tags` and `--wandb-tags` for consistency.

File: rl_zoo3/train.py [hunk 3]
```diff
@@ -206,10 +208,12 @@ def train():
             )
 
         run_name = f"{args.env}__{args.algo}__{args.seed}__{int(time.time())}"
+        tags = [f"v{sb3.__version__}"] + (args.tags if args.tags else [])
```
Comment (araffin) [inline rl_zoo3/train.py:211]: you should be able to simplify this line with the new default

File: CHANGELOG.md [hunk 4]
```diff
@@ -190,6 +192,7 @@
 
 ### New Features
 - Added support for Python 3.10
+- Added
```
Comment (araffin) [inline CHANGELOG.md:195]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,8 @@
 - Added ``monitor_kwargs`` parameter
 - Handle the `env_kwargs` of `render:True` under the hood for panda-gym v1 envs in `enjoy` replay to match visualzation behavior of other envs
 - Added support for python config file
+- Added ``--tags`` argument to ``train.py`` to add tags to the wandb run
```
Comment (araffin) [inline CHANGELOG.md:12]: maybe add an example in the doc (just update the one with wandb)

Comment (qgallouedec) [reply CHANGELOG.md:12]: I would need the command line example in the doc as minimal as possible. So I'd rather add a note after.

**Review by araffin — APPROVED** [review]
LGTM =)

**Pull request merged**